terraform/install.sh

```diff
@@ -201,12 +202,31 @@ applyTerraform() {
     terraform init -backend-config "bucket=${bucket_name}" -lockfile=false # lock-free to prevent access fail
   fi
 
-  log "Apply Terraform automation"
+  #build Terraform apply command 
+  terraform_command="terraform apply -auto-approve -var=\"project_id=${project_id}\" -var=\"bucket_name=${bucket_name}\" -var=\"skip_loadgen=${skip_loadgen:-false}\""
+
+  #If billing account provided specify it 
   if [[ -n "$billing_id" ]]; then
-    terraform apply -auto-approve -var="billing_account=${billing_acct}" -var="project_id=${project_id}" -var="bucket_name=${bucket_name}" -var="skip_loadgen=${skip_loadgen:-false}"
-  else
-    terraform apply -auto-approve -var="project_id=${project_id}" -var="bucket_name=${bucket_name}"  -var="skip_loadgen=${skip_loadgen:-false}"
+    terraform_command+=" -var=\"billing_account=${billing_acct}\"" 
   fi
+  #Check application version
+  #If cluster already exist leave original version
+  gke_location="$(gcloud container clusters list --format="value(location)" --filter name=cloud-ops-sandbox)"
+  if [[ -n "$gke_location" ]]; then 
+    gke_version="$(gcloud container clusters describe cloud-ops-sandbox --region "${gke_location}"  --format="value(resourceLabels.version)")"
+    #If cluster exist and it's older version use backward comp params
+    if [[ -n "$gke_version" ]]; then 
+       app_ver=$gke_version
+    fi
+  elif [[ -n "$VERSION" ]]; then
+    app_ver="$(echo $VERSION | tr "." "_")"
+  else
+    app_ver="0" 
+  fi     
```

Comment:
if `gke_version` is not initialized we may end up with undefined `app_ver`. Maybe it worth to initiate it to `"0"` and then to run if-else logic?